aiomonitor/monitor.py

```diff
@@ -147,13 +133,41 @@ def _server(self) -> None:
             while not self._closing.is_set():
                 try:
                     client, addr = sock.accept()
-                    with client:
-                        sout = client.makefile('w', encoding='utf-8')
-                        sin = client.makefile('r', encoding='utf-8')
-                        self._interactive_loop(sin, sout)
-                except (socket.timeout, OSError):
+                    threading.Thread(target=CLI(client, addr, self)).start()
+                except OSError:
                     continue
 
+
+class CLI:
+    prompt = 'monitor >>> '
+    intro = '\nAsyncio Monitor: {tasknum} task{s} running\nType help for available commands\n\n'  # noqa
+    help_template = '{cmd_name} {arg_list}\n    {doc}\n'
+    help_short_template = '    {cmd_name}{cmd_arg_sep}{arg_list}: {doc_firstline}'  # noqa
+
+    _cmd_prefix = 'do_'
+    _empty_result = object()
+    _sin = None  # type: IO[str]
+    _sout = None  # type: IO[str]
+
+    def __init__(self,
+                 client: socket.socket,
+                 addr: Tuple[str, int],
+                 monitor: Monitor
```

Comment:
I think instead of sharing full monitor object safer to pass required arguments like thread_id/etc, less sharing better

Reply:
Agreed, I had planned on cleaning this up, but for the first impression, I just wanted to make it work.